PittAPI/news.py

```diff
@@ -0,0 +1,61 @@
+'''
+The Pitt API, to access workable data of the University of Pittsburgh
+Copyright (C) 2015 Ritwik Gupta
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 2 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+'''
+
+import re
+import requests
+from bs4 import BeautifulSoup, SoupStrainer
+from requests.packages.urllib3.exceptions import InsecureRequestWarning
+
+requests.packages.urllib3.disable_warnings(InsecureRequestWarning)
+
+sess = requests.session()
+strainer = SoupStrainer('div', attrs={'class': 'kgoui_list_item_textblock'})
+
+
+def get_news(feed="main_news"):
+    # feed indicates the desired news feed
+    # "main_news"      - main news
+    # "cssd"           - student announcements, on my pitt
+    # "news_chronicle" - the Pitt Chronicle news
+    # "news_alerts"    - crime alerts
+
+    # sample_url = 'https://m.pitt.edu/news/index.json?feed=main_news&id=&_object=kgoui_Rcontent_I0_Rcontent_I0&_object_include_html=1'
+    news = [] 
+
+    end_loop = False
+    counter = 0
+    while not end_loop:
+        url = u'https://m.pitt.edu/news/index.json?feed={}&id=&_object=kgoui_Rcontent_I0_Rcontent_I0&_object_include_html=1'.format(feed) + u'&start=' + str(counter)
+        data = sess.get(url).json()  # Should be UTF-8 by JSON standard
+        soup = BeautifulSoup(data['response']['html'], 'lxml') #, parse_only=strainer)
+	news_names = map((lambda i: i.getText()), soup.find_all('span', class_='kgoui_list_item_title'))
+	news_links = map((lambda i: i['href']), soup.find_all('a', class_="kgoui_list_item_action"))
+	news_links = map((lambda i: re.sub(r"\+at\+.+edu", "", i)), news_links)
+	news_links = map((lambda i: i.replace("/news", "https://m.pitt.edu/news")), news_links)
+	news_links = map((lambda i: unicode(i, 'utf-8')), news_links)
```

Comment:
You just need to replace the tabs in front of all the `news_names` and `news_links` with 8 spaces and then it will pass the tests and work correctly. Python doesn't particularly like if you mix tabs and spaces so it's giving an error to force you to choose one, in our case spaces.